### ISSUE TITLE
BatchNorm + Convolution fusion in `prepare_pt2e` removal

### ISSUE THREAD

**Issue by skywall** [pytorch/ao#2245]
### 🚀 The feature, motivation and pitch

There is BatchNorm + Convolution fusion happening in `prepare_pt2e` [function call](https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/quantize_pt2e.py#L95). I don't think it's good practice to introduce optimizations randomly into the pipeline (`prepare_pt2e` function call) because this might be a side effect for the user of an API. 

I'm proposing removal of this optimization and letting user peform this optimization on his own, before quantization.

Follow-up from: https://github.com/pytorch/executorch/pull/10579#issuecomment-2899871340
cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @msaroufim @digantdesai
